docs/gallery/domain/ecephys.py

```diff
@@ -5,152 +5,291 @@
+# This is analogous to how we can store the :py:class:`~pynwb.behavior.SpatialSeries` object inside of a
+# :py:class:`~pynwb.behavior.Position` object.
+#
+# .. only:: html
+#
+#   .. image:: ../../_static/LFP.svg
+#     :width: 800
+#     :alt: LFP UML diagram
+#     :align: center
+#
+# .. only:: latex
+#
+#   .. image:: ../../_static/LFP.png
+#     :width: 800
+#     :alt: LFP UML diagram
+#     :align: center
+#
+
+lfp = LFP(electrical_series=lfp_electrical_series)
+
+####################
+# Unlike the raw data, which we put into the acquisition group of the :py:class:`~pynwb.file.NWBFile`,
+# LFP data is typically considered processed data because the raw data was filtered and downsampled to generate the LFP.
+#
+# Create a processing module named ``"ecephys"`` and add the :py:class:`~pynwb.ecephys.LFP` object to it.
+# This is analogous to how we can store the :py:class:`~pynwb.behavior.Position` object in a processing module
+# with the :py:class:`~pynwb.file.NWBFile.create_processing_module` method.
+
+
+ecephys_module = nwbfile.create_processing_module(
+    name='ecephys',
+    description='processed extracellular electrophysiology data'
+)
+ecephys_module.add(lfp)
 
 ####################
 # .. _units_electrode:
 #
-# Associate electrodes with units
-# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Spike Times
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 #
-# The :ref:`PyNWB Basics tutorial <basics>` demonstrates how to add data about units and specifying custom metadata
-# about units. As mentioned :ref:`here <units_fields_ref>`, there are some optional fields for units, one of these
-# is *electrodes*. This field takes a list of indices into the electrode table for the electrodes that the unit
-# corresponds to. For example, if two units were inferred from the first electrode (*id* = 1, index = 0), you would
-# specify that like so:
+# Spike times are stored in the :py:class:`~pynwb.misc.Units` table, which is another subclass of :py:class:`~hdmf.common.table.DynamicTable`.
+# Adding columns to the :py:class:`~pynwb.misc.Units` table is analogous to how we can add columns to the ``"electrodes"`` and ``"trials"`` tables.
+#
+# We will generate some random spike data and populate the :py:class:`~pynwb.misc.Units` table using the :py:class:`~pynwb.file.NWBFile.add_unit`
+# method. Then we can display the :py:class:`~pynwb.misc.Units` table as a pandas :py:class:`~pandas.DataFrame`.
+
+nwbfile.add_unit_column(name='quality', description='sorting quality')
 
-nwbfile.add_unit(id=1, electrodes=[0])
-nwbfile.add_unit(id=2, electrodes=[0])
+poisson_lambda = 20
+firing_rate = 20
+n_units = 10
+for n_units_per_shank in range(n_units):
+    n_spikes = np.random.poisson(lam=poisson_lambda)
+    spike_times = np.round(np.cumsum(np.random.exponential(1 / firing_rate, n_spikes)), 5)
+    nwbfile.add_unit(spike_times=spike_times,
+                     quality='good',
+                     waveform_mean=[1., 2., 3., 4., 5.])
+
+nwbfile.units.to_dataframe()
```

Comment:
When I reviewed the changes with `make htmldoc`, the dataframe is not showing up.
Any suggestions how to make this part how the spike times dataframe?

Reply:
I believe the file must start with the prefix "plot_". @oruebel  is this correct?

Reply:
Yes, renaming the tutorial file to start with ``plot_`` is the simplest way to fix this issue. Sphinx gallery uses the name as an indicator that it should render also the output of code cells in the docs. Note, this will render the output of all cells, i.e., you may need to do some clean-up once you enable the option to make sure only the cells that you want to show generate output.